src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/MetricItemExtensions.cs

```diff
@@ -379,51 +355,48 @@ private static void AddScopeAttributes(IEnumerable<KeyValuePair<string, object>>
         }
     }
 
-    /*
-    [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    private static OtlpMetrics.Exemplar ToOtlpExemplar(this IExemplar exemplar)
+    private static OtlpMetrics.Exemplar ToOtlpExemplar<T>(T value, in Metrics.Exemplar exemplar)
+        where T : struct
     {
-        var otlpExemplar = new OtlpMetrics.Exemplar();
-
-        if (exemplar.Value is double doubleValue)
+        var otlpExemplar = new OtlpMetrics.Exemplar
         {
-            otlpExemplar.AsDouble = doubleValue;
-        }
-        else if (exemplar.Value is long longValue)
+            TimeUnixNano = (ulong)exemplar.Timestamp.ToUnixTimeNanoseconds(),
+        };
+
+        if (exemplar.TraceId.HasValue)
         {
-            otlpExemplar.AsInt = longValue;
+            byte[] traceIdBytes = new byte[16];
+            exemplar.TraceId.Value.CopyTo(traceIdBytes);
+
+            byte[] spanIdBytes = new byte[8];
+            exemplar.SpanId.Value.CopyTo(spanIdBytes);
+
+            otlpExemplar.TraceId = UnsafeByteOperations.UnsafeWrap(traceIdBytes);
+            otlpExemplar.SpanId = UnsafeByteOperations.UnsafeWrap(spanIdBytes);
         }
-        else
+
+        if (typeof(T) == typeof(long))
         {
-            // TODO: Determine how we want to handle exceptions here.
-            // Do we want to just skip this exemplar and move on?
-            // Should we skip recording the whole metric?
-            throw new ArgumentException();
+            otlpExemplar.AsInt = (long)(object)value;
         }
-
-        otlpExemplar.TimeUnixNano = (ulong)exemplar.Timestamp.ToUnixTimeNanoseconds();
-
-        // TODO: Do the TagEnumerationState thing.
-        foreach (var tag in exemplar.FilteredTags)
+        else if (typeof(T) == typeof(double))
         {
-            otlpExemplar.FilteredAttributes.Add(tag.ToOtlpAttribute());
+            otlpExemplar.AsDouble = (double)(object)value;
```

Comment:
Do we need the casting here? `value` is already a `double` type.

Reply:
It is needed ya. It is a quirk of generics. 

```diff
        else if (typeof(T) == typeof(double))
        {
-            otlpExemplar.AsDouble = (double)(object)value;
+            otlpExemplar.AsDouble = value; // Error CS0029 Cannot implicitly convert type 'T' to 'double'
        }
```

But thankfully it isn't anything to worry about! The JIT will do the right thing and elide the cast to `object` when it specializes for `T`s of `double`: [SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgBEBLbAcwDsJcGDmFwBuGvQYBlABbYoABwAy2YCwBKAV15D8McdQkBmJqQYBhBgG8aDOwwDaAKQ4YA4jF4wowgBQYATwUYCAAzXwATCE1gABsYAEoEgF1be2ITZiQGKJj4hgAVCDZouJgAHgKAPl8ChgA3bFjNRIYAdxlvGEKGEAZBKE0wDGs0+3sOUIZ/IJDwgoSGAF4lhkDgsMjS+KSx8dHqfaOmAHZp3LKE3whgACsYYYTG5v098YBfN/sYWNwYL7sNkOx3GbBgwE03BYADFsBxYr4AEQAVS8CGCwxgETWs0RCQMIPGxDO5ggvHq3gwLCKJTyMF8zxa+IBDHeIL2n2Bdj2GSY5GyF3yNO29MF3UZiT2QJBxIaTRaBPsnPeQA==)